libs/partners/pinecone/langchain_pinecone/embeddings.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+from typing import Dict, Iterable, List, Optional
+
+import aiohttp
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import (
+    BaseModel,
+    Extra,
+    Field,
+    SecretStr,
+    root_validator,
+)
+from langchain_core.utils import convert_to_secret_str
+from pinecone import Pinecone as PineconeClient  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class PineconeEmbeddings(BaseModel, Embeddings):
+    """PineconeEmbeddings embedding model.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_pinecone import PineconeEmbeddings
+
+            model = PineconeEmbeddings()
+    """
+
+    _client: PineconeClient = Field(exclude=True)
+    _async_client: aiohttp.ClientSession = Field(exclude=True)
+    model: str
+    batch_size: int
+    show_progress_bar: bool = False
+    truncation: Optional[bool] = None
+    pinecone_api_key: Optional[SecretStr] = None
+
+    class Config:
+        extra = Extra.forbid
+
+    @root_validator(pre=True)
+    def default_values(cls, values: dict) -> dict:
+        """Set default batch size"""
+
+        batch_size = values.get("batch_size")
+        if batch_size is None:
+            values["batch_size"] = 128
```

Comment:
The max batch size we currently support is 96 for the e5-large model. Not sure if 128 is right here.

Reply:
will update, thanks!

Reply:
^ done